packages/sdk/src/util/userFilter.ts

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ActionHandler, Guid, User } from '..';
+
+/** A callback that handles when a user joins or leaves the MRE session */
+export type UserEntryExitCallback = (user: User) => void;
+
+/** An event type for [[UserFilter.shouldForwardUserEvent]] */
+export type UserInteractionType = 'joined' | 'input';
```

Comment:
>input [](start = 46, length = 5)

Is this just here to support the potential need in differentiating if something is an input event in a concrete implementation of the UserFilter predicate method?  I don't see where we ever check against it in the filters implemented.  Just curious if that is the intention.

Reply:
The moderator filter uses this, so moderators can only be added at join and not on click.

---
In reply to: [446271464](https://github.com/microsoft/mixed-reality-extension-sdk/pull/634#discussion_r446271464) [](ancestors = 446271464)